Update linux installation instruction (#90)
getml/getml-docs: docs/install/packages/linux.md

## Separate installation of Engine {#separate-installation-of-engine}

In some cases, it might be preferred to install the Engine separately on Linux
using [CLI][cli]. For example, if you want to use the [Enterprise
edition][enterprise-benefits] of the Engine. You can download the current

> release's package from the [download page][packge-download].

Reviewer: ```suggestion
release's package from the [download page][package-download].
```

If you want to download getML from the command line, you can use the following
commands. Replace `<arch>` with either `amd64` or `arm64`, depending on your
architecture. You can use `uname -m` to figure out the architecture. If it
returns `aarch64` or `arm64`, you need to use `arm64`, otherwise go with
`amd64`.

```bash
curl -LO https://static.getml.com/download/<version>/getml-community-<version>-<arch>-linux.tar.gz

# If you want to check the hash:
# curl -LO https://static.getml.com/download/<version>/getml-community-<version>-<arch>-linux.tar.gz.sha256
# if [ "$(sha256sum getml-community-<version>-<arch>-linux.tar.gz) == $(cat getml-community-<version>-<arch>-linux.tar.gz.sha256)" ]; then echo "OK"; else echo "NOT OK"; fi

tar -xzf getml-community-<version>-<arch>-linux.tar.gz

> getml-community-<version>-<arch>-linux/getML install

Author: ```suggestion
cd getml-community-<version>-<arch>-linux
./getML install
```

## Separate installation of Engine {#separate-installation-of-engine}

In some cases, it might be preferred to install the Engine separately on Linux
using [CLI][cli]. For example, if you want to use the [Enterprise
edition][enterprise-benefits] of the Engine. You can download the current

> release's package from the [download page][package-download].

Author: @jan-meyer-1986, what do you think?

If you want to download getML from the command line, you can use the following
commands. Replace `<arch>` with either `amd64` or `arm64`, depending on your
architecture. You can use `uname -m` to figure out the architecture. If it
returns `aarch64` or `arm64`, you need to use `arm64`, otherwise go with
`amd64`.

```bash
curl -LO https://static.getml.com/download/<version>/getml-community-<version>-<arch>-linux.tar.gz

# If you want to check the hash:
# curl -LO https://static.getml.com/download/<version>/getml-community-<version>-<arch>-linux.tar.gz.sha256

> # if [ "$(sha256sum getml-community-<version>-<arch>-linux.tar.gz) == $(cat getml-community-<version>-<arch>-linux.tar.gz.sha256)" ]; then echo "OK"; else echo "NOT OK"; fi

Author: ```suggestion
# if [ "$(sha256sum getml-community-<version>-<arch>-linux.tar.gz)" == "$(cat getml-community-<version>-<arch>-linux.tar.gz.sha256)" ]; then echo "OK"; else echo "NOT OK"; fi
```

In some cases, it might be preferred to install the Engine separately on Linux using [CLI][cli]. For example, if you want to use the [Enterprise edition][enterprise-benefits] of the Engine.

> Please execute the following commands, replacing `<arch>` with either `amd64` or `arm64`, depending on your architecture.

Reviewer 2: I have always wanted to introduce -on the spot- code adjustments, just like google does (e.g.  like 'PROJECT_ID' in https://cloud.google.com/run/docs/quickstarts/build-and-deploy/deploy-go-service?hl=en). This eases usage remarkably. But I never figured out, how to do it.

Reviewer: It might be possible with some JavaScript, according to ChatGPT: https://drive.google.com/file/d/15VfcArUBDKzznDuiGSkvZmfZUMnC4Zwt/view?usp=drive_link

I haven't tried it, tough.

If you are unsure, `amd64` is probably the right choice.
You can also use `uname -m` to figure out the architecture.
If it says something like `aarch64` or `arm64`, you need to use `arm64`, otherwise go with `amd64`.

```bash

> curl -L https://static.getml.com/download/<version>/getml-community-<version>-<arch>-linux.tar.gz

Reviewer 2: I am getting:
```
Warning: Binary output can mess up your terminal. Use "--output -" to tell 
Warning: curl to output it to your terminal anyway, or consider "--output 
Warning: <FILE>" to save to a file.
```
then I ran 
```
jan-meyer@jan:~/Downloads$ curl -L https://static.getml.com/download/1.5.0/getml-community-1.5.0-amd64-linux.tar.gz --output getml-community-1.5.0-amd64-linux.tar.gz
```

Author: Thanks, `-O` is missing. Will add tomorrow.

Reviewer: The `curl` command for downloading the hash also needs `-o`.

# If you want to check the hash:
# curl -L https://static.getml.com/download/<version>/getml-community-<version>-<arch>-linux.tar.gz.sha256
# if [ "$(sha256sum getml-community-<version>-<arch>-linux.tar.gz) == $(cat getml-community-<version>-<arch>-linux.tar.gz.sha256)" ]; then echo "OK"; else echo "NOT OK"; fi

> tar -xzf getml-community-<version>-<arch>-linux.tar.gz

Author: Is the tarball actually there?

> tar -xzf getml-community-<version>-<arch>-linux.tar.gz

Reviewer 2: yes

> tar -xzf getml-community-<version>-<arch>-linux.tar.gz

Author: Let's debug tomorrow.

> tar -xzf getml-community-<version>-<arch>-linux.tar.gz

Reviewer: Tarball extraction is working for me.

![image](https://github.com/user-attachments/assets/b9a88e80-035d-433b-8b16-f88ab4e3391a)

> getml-community-<version>-<arch>-linux/getML install

Reviewer 2: and subsequently this fails

Reviewer 2: my resulting folder is nested, I need to do :
`getml-community-1.5.0-amd64-linux/getml-community-1.5.0-amd64-linux/getML install`

Reviewer: Same for `arm64` tarball.

Author: This is fixed on monorepo!471.

```bash

> ./getML

Reviewer: If we remove `cd getml-community-<version>-<arch>-linux`, then for execution of the engine, it should be `getml-community-<version>-<arch>-linux/getML`? 

I feel we should keep `cd getml-community-<version>-<arch>-linux` and then it will be simplier: `./getML` (for execution) and `./getML install` (for installation).

Author: I guess this is fixed now?

Reviewer 3: Yes.

Reviewer: Yes

```

> If a global installation is sucessful, getML is added to `/usr/local/bin` which should be on already on `PATH` on every distribution. If the Engine was installed to the user home directory, you can add the installation directory to your `PATH` variable if you want to call the getML [CLI][cli] from anywhere.

Reviewer 2: needs clarification: which should be on already on `PATH` on every distribution.

./getML
```

> If a global installation is successful, getML is added to `/usr/local/bin` which
> should be on already on `PATH` on every distribution. If the Engine was
> installed to the user home directory, you can add the installation directory to
> your `PATH` variable if you want to call the getML [CLI][cli] from anywhere.

Author: ```suggestion
If you install getML globally and the installation is successful, the `getML`
executable is placed in the directory `/usr/local/bin`. This directory is
typically included in the `PATH` environment variable on most major Linux
distributions, so you should be able to run `getML` from any terminal session
without any additional configuration.

However, if you installed the getML Engine in your user home directory rather
than globally, it's stored in a location that isn't automatically included in
the `PATH`. In this case, you can add the installation directory to your `PATH`
variable to to be able to call the getML CLI from any terminal session.
```

Author: @jan-meyer-1986, okay?

Reviewer 2: variable to to be able

Reviewer 2: "variable to to be able" -> "variable to be able"
